auslib/web/admin/views/rules.py

```diff
@@ -320,42 +318,68 @@ def _post(self, transaction, changed_by):
             for field in ["update_type", "backgroundRate", "priority"]:
                 if not what.get(field, None):
                     return problem(400, "Bad Request", "Missing field", ext={"exception": "%s is missing" % field})
-        else:
-            return problem(400, "Bad Request", "Invalid or missing change_type")
 
         if change_type in ["update", "insert"]:
             rule_dict, mapping_values, fallback_mapping_values = process_rule_form(what)
             what = rule_dict
             if what.get('mapping') is not None and len(mapping_values) != 1:
                 return problem(400, 'Bad Request', 'Invalid mapping value. No release name found in DB')
 
-            if what.get('fallbackMapping') is not None and len(fallback_mapping_values) != 1:
+            elif what.get('fallbackMapping') is not None and len(fallback_mapping_values) != 1:
                 return problem(400, 'Bad Request', 'Invalid fallbackMapping value. No release name found in DB')
 
         return super(RuleScheduledChangesView, self)._post(what, transaction, changed_by, change_type)
 
 
 class RuleScheduledChangeView(ScheduledChangeView):
+    """/scheduled_changes/rules/<int:sc_id>"""
+
     def __init__(self):
         super(RuleScheduledChangeView, self).__init__("rules", dbo.rules)
 
     @requirelogin
     def _post(self, sc_id, transaction, changed_by):
-        if connexion.request.get_json() and connexion.request.get_json().get("data_version"):
-            if connexion.request.get_json().get("change_type") == "delete":
-                form = EditScheduledChangeDeleteRuleForm()
-            else:
-                form = EditScheduledChangeExistingRuleForm()
+        # TODO: modify UI and clients to stop sending 'change_type' in request body
+        sc_rule = self.sc_table.select(where={"sc_id": sc_id}, transaction=transaction, columns=["change_type"])
+        if sc_rule:
+            change_type = sc_rule[0]["change_type"]
         else:
-            form = EditScheduledChangeNewRuleForm()
+            return problem(404, "Not Found", "Unknown sc_id",
+                           ext={"exception": "No scheduled change for rule found for given sc_id"})
+
+        what = {}
+        for field in connexion.request.get_json():
+            # Unlike when scheduling a new change to an existing rule, rule_id is not
+            # required (or even allowed) when modifying a scheduled change for an
+            # existing rule. Allowing it to be modified would be confusing.
+            if (field in ["csrf_token", "rule_id"] or (change_type == "insert" and field == "data_version") or
```

Comment:
As discussed over video, ignore this - I was getting confused between creating and editing scheduled changes (rule_id is only required for the former).